cuda_core/cuda/core/experimental/_event.pyx

```diff
@@ -4,28 +4,28 @@
 
 from __future__ import annotations
 
-import weakref
+from cuda.core.experimental._utils.cuda_utils cimport (
+    _check_driver_error as raise_if_driver_error,
+    check_or_create_options,
+)
+
 from dataclasses import dataclass
 from typing import TYPE_CHECKING, Optional
 
 from cuda.core.experimental._context import Context
 from cuda.core.experimental._utils.cuda_utils import (
     CUDAError,
-    check_or_create_options,
     driver,
     handle_return,
 )
-from cuda.core.experimental._utils.cuda_utils import (
-    _check_driver_error as raise_if_driver_error,
-)
 
 if TYPE_CHECKING:
     import cuda.bindings
     from cuda.core.experimental._device import Device
 
 
 @dataclass
-class EventOptions:
+cdef class EventOptions:
```

Comment:
Note to self: It would be interesting to compare generated C++ code between using `@dataclass` and using `@cython.dataclasses.dataclass`.

Since the latter does not require `import dataclasses`, it might result in leaner code.

Reply:
Yeah would be interesting to confirm Cython's claim that both are equivalent
https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#dataclass-extension-types